src/features/spa/aggregate/interaction-node.js

```diff
@@ -67,11 +67,23 @@ InteractionNodePrototype.finish = function finish (timestamp) {
   var node = this
   if (node.end) return
   node.end = timestamp
-  var parent = node.parent
-  while (parent.cancelled) parent = parent.parent
-  parent.children.push(node)
-  node.parent = null
 
+  // Find the next parent node that is not cancelled
+  if (node.parent) {
+    let parent = node.parent
+    while (parent.cancelled) {
```

Comment:
Just this would work, right?
```suggestion
    while (parent?.cancelled) {
```

Reply:
No because parent could still get set to a null value in the previous loop.

Reply:
Actually I'm suggesting this change instead overall:
```suggestion
    let parent = node.parent;
    while (parent?.cancelled) parent = parent.parent; // keep searching up the chain until we find a/the first ancestor that isn't cancelled
    if (parent) parent.children.push(node); // if parent isn't null, then we found one that isn't cancelled
    node.parent = null;
```

Reply:
I'm also quite confused by the original code, since if we _do_ find a cancelled ancestor, we say `ancestor -> node` but is why is `node.parent -> null` ?? Lol, any ideas?

Reply:
Yea that would work to. Ill commit that.

Reply:
> No because parent could still get set to a null value in the previous loop.

If `parent = null` from previous loop, then `parent?.cancelled` would be `undefined` and fail to continue looping though.

Reply:
> I'm also quite confused by the original code, since if we _do_ find a non-cancelled ancestor, we say `ancestor.children -> node` but is why is `node.parent -> null` ?? Lol, any ideas?

No, Jordan and I were confused by that too this morning when we were working through this.

Reply:
Maybe this is a bug... possible they meant `node.parent = parent`, whether that be a found node or `null`. "The world may never know."